app/javascript/pages/Settings/Payments/Show.tsx

```diff
@@ -223,53 +129,74 @@ const PaymentsPage = (props: Props) => {
       } else if (newPayoutMethod === "bank") {
-        updateComplianceInfo({ is_business: true });
+        form.setData("user", { ...form.data.user, is_business: true });
       }
     }
   };
 
-  const [payoutsPausedByUser, setPayoutsPausedByUser] = React.useState(props.payouts_paused_by_user);
-
-  const [payoutThresholdCents, setPayoutThresholdCents] = React.useState<{ value: number | null; error?: boolean }>({
-    value: props.payout_threshold_cents,
-  });
-  const [payoutFrequency, setPayoutFrequency] = React.useState<PayoutFrequency>(props.payout_frequency);
-
-  const [complianceInfo, setComplianceInfo] = React.useState(props.compliance_info);
   const updateComplianceInfo = (newComplianceInfo: Partial<ComplianceInfo>) => {
     if (
       props.user.country_code &&
       newComplianceInfo.updated_country_code &&
       props.user.country_code !== newComplianceInfo.updated_country_code
     ) {
-      setErrorMessage(null);
       setIsUpdateCountryConfirmed(false);
       setShowUpdateCountryConfirmationModal(true);
     }
-    setComplianceInfo((prevComplianceInfo) => ({ ...prevComplianceInfo, ...newComplianceInfo }));
+    form.setData("user", { ...form.data.user, ...newComplianceInfo });
     setErrorFieldNames(new Set());
   };
 
-  const [bankAccount, setBankAccount] = React.useState(props.bank_account_details.bank_account);
   const updateBankAccount = (newBankAccount: Partial<BankAccount>) => {
-    setBankAccount((prevBankAccount) => ({ ...prevBankAccount, ...newBankAccount }));
+    form.setData("bank_account", { ...form.data.bank_account, ...newBankAccount });
     setErrorFieldNames(new Set());
   };
 
-  const [paypalEmailAddress, setPaypalEmailAddress] = React.useState(props.paypal_address);
   const [debitCard, setDebitCard] = React.useState<PayoutDebitCardData | null>(null);
   const [showNewBankAccount, setShowNewBankAccount] = React.useState(!props.bank_account_details.account_number_visual);
+  const previousCountryRef = React.useRef<string | null>(
+    props.compliance_info.is_business ? props.compliance_info.business_country : props.compliance_info.country,
+  );
+
+  // Reset form data when country changes
+  React.useEffect(() => {
+    const currentCountry = props.compliance_info.is_business
+      ? props.compliance_info.business_country
+      : props.compliance_info.country;
+
+    if (previousCountryRef.current !== currentCountry) {
+      form.setData({
+        user: props.compliance_info,
+        payouts_paused_by_user: props.payouts_paused_by_user,
+        payout_threshold_cents: props.payout_threshold_cents,
+        payout_frequency: props.payout_frequency,
+        bank_account: props.bank_account_details.bank_account,
+        payment_address: props.paypal_address,
+      });
+      setErrorFieldNames(new Set());
+      setClientErrorMessage(null);
+      previousCountryRef.current = currentCountry;
+    }
+  }, [props.compliance_info]);
```

Comment:
We need to reset the entire form when the country changes and only when the country changes, the previous code caused this CI [error](https://github.com/antiwork/gumroad/actions/runs/20246240294/job/58127433328?pr=2331) which is fixed now